server/client_fake.go

```diff
@@ -39,13 +39,18 @@ func (client *ClientFake) NodeSendStats(nodeKey string, sessionStats []dto.Sessi
 
 func (client *ClientFake) FindProposals(nodeKey string) (proposals []dto_discovery.ServiceProposal, err error) {
 	log.Info(MYSTERIUM_API_LOG_PREFIX, "Fake proposals requested for node_key: ", nodeKey)
-	if proposal, ok := client.proposalsByProvider[nodeKey]; ok {
-		proposals = []dto_discovery.ServiceProposal{proposal}
-	} else {
-		proposals = []dto_discovery.ServiceProposal{}
+
+	for _, proposal := range client.proposalsMock {
+		var filterMatched = true
+		if nodeKey != "" {
```

Comment:
What about moving this check outside loop and making like this:
if (nodeKey == "") return client.proposalsMock

Reply:
I did not wanted conditional code with&without filter

Reply:
And now you have conditional code with and without filter inside a loop

Reply:
@tadovas this is client mock, do you think its important to optimize here?

Reply:
Well it's not about optimization, more about readability. And yes you are right - since it's just for testing - no point to do some optimizations here, we can focus more on readability.

Reply:
Plus, it's extendable - if we add more filters, it will be easy to extend this method by adding more ifs inside loop :)